### ISSUE TITLE
Results are invalid because you use different claim-extraction models for fradulent and legitimate documents

### ISSUE THREAD

**Issue by josh-gleason** [GIFRN/Scientific-Fraud-Detection#1]
I discovered that the code is fundamentally flawed in that it uses completely different claim-extraction models, with very different characteristics, to classify the fraudulent data and the legitimate data. Fixing this results in significantly degraded performance.

You split the test set into "fraudulent" and "legitimate" and for some reason use different claim extraction models on each split (fraudulent: [models/argumentation_based_fraud_detection.py#L250-L348](https://github.com/GIFRN/Scientific-Fraud-Detection/blob/main/models/argumentation_based_fraud_detection.py#L250-L348) and legitimate: [models/argumentation_based_fraud_detection.py#L352-L450](https://github.com/GIFRN/Scientific-Fraud-Detection/blob/main/models/argumentation_based_fraud_detection.py#L352-L450)).

Assuming the same **hard coded random seeds** you use, this leads to two different claim extraction models with extremely different characteristics (because the training of that model is quite unstable). In the first training, it just so happens that the **claim extractor largely collapses**, predicting "none" for nearly every sentence. This leads to about 60% of abstracts becoming empty strings. The next time you retrain the claim extraction, it doesn't completely collapse, and produces empty abstracts only about 12% of the time. Because empty strings are strongly classified as fraudulent, so your script reports high accuracy, though it's fundamentally invalid.

- If I try using the same claim extraction model for both fraudulent and legitimate data then I get poor performance 
- If I try reversing the order of the two training loops then I get poor performance
- If I change the hard coded random seed I get terrible performance

Could you clarify the rationale in training two different claim-extraction models? Please help me understand if I'm mistaken, but it seems to me that the published results of this work are likely invalid.

**Comment by GIFRN** [comment]
Thanks for taking an interest in the work! The rationale is that these experiments were done using (relatively) old language models, which are not capable of generalising to any great degree OOD (even between legitimate and illegitimate samples). For a more up to the date approach I can point you towards this paper: https://aclanthology.org/2025.coling-main.569/.

I think your concern about the robustness of the results is a valid concern and something that we should've experimented with at the the time/addressed in the paper. However the paper makes a weaker claim about how these features could provide _some_ signal under certain circumstances.

This is a more recent attempt trying to address this problem from a different angle: https://github.com/tigerchen52/pub_guard_llm. Please let me know if you have any more questions!

**Comment by josh-gleason** [comment]
The issue is not redundancy. The code selects different claim-extractors based on the true test label, which is label leakage. This makes reported accuracy invalid. Using a single claim extractor trained once removes the leakage and the performance collapses. If this is the code that produced the experimental results reported in your manuscript, then there should be a warning at the top of your repo to warn others that the methodology in this repository is not valid.
